src/main/java/Duke.java

```diff
@@ -1,10 +1,181 @@
+                } else if (curTask.equals("todo")) {
+                    addTodo(echo, tasks, index);
+                } else if (curTask.equals("delete")) {
+                    deleteTask(tasks, index);
+                    index = index - 2;
+                }
+                index++;
+                writeToFile(tasks);
+                System.out.print("Now you have ");
+                System.out.println(index + " tasks in the list.");
+                System.out.println("----------------------------------------------------------------");
+
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-( ");
+                System.out.println("----------------------------------------------------------------");
+            } catch (NullPointerException e) {
+                System.out.println("☹ OOPS!!! I'm sorry, but seems like there is no such task :-( ");
+                System.out.println("----------------------------------------------------------------");
+            } catch (ChangeStatusException e) {
+                System.out.println("☹ OOPS!!! I'm sorry, but seems like this task is already marked/unmarked");
+                System.out.println("----------------------------------------------------------------");
+            } catch (DukeException e) {
+                System.out.print("☹ OOPS!!! The description of a ");
+                System.out.print(e);
+                System.out.println(" cannot be empty");
+                System.out.println("----------------------------------------------------------------");
+            }
+        }
+    }
+
+    private static void writeToFile(ArrayList<Task> tasks) throws IOException {
+        FileWriter fw = new FileWriter("duke.txt");
+        for(Task t: tasks){
+            fw.write(t.toString()+"\n");
+        }
+        fw.close();
+    }
+
+    private static void printFileContents(String filePath) throws IOException, FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            System.out.println(s.nextLine());
+        }
+    }
+
+
+    public static void deleteTask(ArrayList<Task> tasks, int index) {
+        System.out.println("Noted. I've removed this task: ");
+        System.out.println(tasks.get(index - 1));
+        tasks.remove(index - 1);
+
+    }
+
+    public static void addTodo(String echo, ArrayList<Task> tasks, int index) {
+        String description = echo.substring(echo.indexOf(" ") + 1);
+        tasks.add(new Todo(description));
+        System.out.println("Got it. I've added this task:  ");
+        System.out.println(tasks.get(index));
+    }
+
+
+    public static void addEvent(String echo, ArrayList<Task> tasks, int index) {
+
+        String event = echo.substring(echo.indexOf("/") + 4);
```

Comment:
`4` is [magic number](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers) which can be replaced with a named constant for better code quality.